os/fs/driver/mtd/smart.c

```diff
@@ -1005,6 +1033,48 @@ static int smart_setsectorsize(FAR struct smart_struct_s *dev, uint16_t size)
 			dev->availSectPerBlk = dev->sectorsPerBlk;
 		}
 	}
+	
+#ifdef CONFIG_MTD_SMART_JOURNALING
+	/** Journal Sector is reserved at the last of smartfs partition, it doesn't use MTD Header.
+	  * Journal sectors divided to 2 area, this will be toggled if anyone is filled with data,
+	  * erase previous one. So total number of journal sector must be N*Eraseblock
+	  * We will use it as a contigous memory space...
+	  */
+	size_t log_size = sizeof(struct smart_journal_logging_data_s);
+	/** First get ideal size of journal sector. Formula is Below (x : data sectors y : journal sectors)
+	  * 1. dev->sectorsize : dev->njournalsectors = x : y  
```

Comment:
Also, if this formula is valid, isn't **y** (journal sectors) and **dev->njournalsectors** the same?

Reply:
Not same, because it is flash, we should align with size of block. so I added below code
```
	dev->njournaleraseblocks = nsector / dev->sectorsPerBlk;

	/* If number of sector is greater than n * eraseblock, adjust it */
	if (nsector % dev->sectorsPerBlk != 0) {
		dev->njournaleraseblocks++;
	}

	/* We have to use 2N of eraseblock for journal, adjust it */
	if (dev->njournaleraseblocks % 2 != 0) {
		dev->njournaleraseblocks++;
	}
```
Point is that 2n * Block is best.

Reply:
Cleared after you gave the above comment.